datasets/leukemia.py

```diff
@@ -16,8 +18,10 @@ def get_data(self):
         # Unlike libsvm[leukemia], this dataset corresponds to the whole
         # leukemia  data with train + test data (72 samples) and not just
         # the training set.
-        X, y = fetch_openml("leukemia", return_X_y=True)
-        X = X.to_numpy()
-        y = LabelBinarizer().fit_transform(y)[:, 0].astype(X.dtype)
+        self.X, self.y = fetch_openml("leukemia", return_X_y=True)
+        self.X = self.X.to_numpy()
+        self.y = LabelBinarizer().fit_transform(self.y)[:, 0].astype(self.X.dtype)
+
+        self.X, self.y = preprocess_data(self.X, self.y)
 
-        return dict(X=X, y=y)
+        return dict(X=self.X, y=self.y)
```

Comment:
Why do you put this in the object? I it is not necessary, I would advise against it, so it does not mess up with hashing and serialization (it should no but we are never too careful :) )

Reply:
Thanks for pointing it out, I fixed it in ebb4ba4.